### ISSUE TITLE
Personal performance rationale

### ISSUE THREAD

**Issue by nilababaei** [nilababaei/portfolio#3]
@thomasjbradley

| | 0 points | 1 points | 2 points | 3 points |
| --- | --- | --- | --- | --- |
| **Patterns** | Barely started | Missing lots of patterns | Has most of the patterns, some missing | **All patterns exist and are well done** |
| **Personality** | Bland, boring, just another graphic designer | Fairly generic but may be better with visual design | Quality layouts and unique, personable content | **Unique and recognizable as you, with engaging, personable content** |
| **Text content** | Fake content only | Some real content, some fake | All real content but with lots of errors | **Real content, well written, no grammar or spelling errors** |
| **Responsiveness** | Not responsive | Works on some screens | Responsive but with lots of awkwardness | **Looks great on all screen sizes** |
| **Documentation** | No documentation | Very little documentation | Mostly well documented | **Amazing! Anybody could understand** |
| **Use of systems** | Everything is repeated | Some good use of Jekyll & Patternbot | Lots of good systemization & reuse | **Lego!** |
| **Code quality** | Barely started | Indentation is barely existent, lots of validation errors, very poor semantics | Decent indentation, just a couple validation errors, decent semantics | **Well indented, fully valid, good semantics** |

Overall, I believe that I should receive an A+ on my pattern library. 

I have been able to design and develop a fully responsive pattern library that showcases my brand through the use of colour and design. Every element on the pattern library is functional and can be paired with any other element. I successfully used all the knowledge that I have learned in the past two years in the web development course to create all the elements I need for my portfolio website through the use of HTML, CSS, and Javascript. I was able to successfully implement the Jekyll page builder features and use skills such as working with Terminal and implementing the Ruby programming language. Through documentation, I have been able to show my use of the Markdown and YAML language as well.

The website as a whole communicates my brand successfully. Through research and brainstorming, I was able to define who I am as a person and figure out ways to implement that into my website. With careful consideration of colour and content, I have showcased my definition of who I am by using a monochromatic colour scheme with an accent yellow. The yellow represents me as a person and my values as it is a colour that exudes happiness, positivity, energy, enlightenment, intellect, and joy. These are all very important qualities that I want future clients to see.

I do not believe that I created any patterns that can not be utilized in my future portfolio website. All of my patterns have been created with a clear and concise coding process that is easily understandable through the effective choice of naming my classes. My code is clean and clear and can be easily adjusted in the future.

An aspect of my pattern library that can be improved is the use of Javascript. As I am moving into a career as a front-end developer, it is important to me that I showcase my abilities throughout my website. This includes Javascript, and I do have one pattern in my library that includes Javascript. In the future, I do have plans to continue adding to my pattern library in ways that include using Javascript for animations and to create more interesting and dynamic patterns.

I did not run into any major problems throughout the process of developing my pattern library. The few issues I faced included problems with the hosting service—Netfliy—and the responsively of the pattern which includes the tabbed interfaces I developed using Javascript. I solved my hosting service issue by using Terminal to remove and reinstall Bundler, which allowed for my Netfliy to sync with the current version of Bundler and Ruby that it uses. For the issue involving my tabbed interfaces, this was solved through looking through my Javascript code which allowed us to see that I had duplicated a section of the code which wasn’t allowing the action to run properly. By removing the duplicated section of code, the tabbed interface pattern began to work as it should.

In conclusion, I believe that I have successfully designed a fully responsive pattern library that communicates who I am through branding and design. The website is usable in terms of functioning as a set of patterns that I can later on place together to form my pages. All the patterns include Jekyll code so I can replace any content or imagery to match the case study related. The pattern library also includes patterns that I can use to add on to the design in the future if need be.
